tools/ci_connector_ops/ci_connector_ops/pipelines/publish.py

```diff
@@ -255,11 +254,11 @@ def create_connector_report(results: List[StepResult]) -> ConnectorReport:
 
             # If the connector image already exists, we don't need to build it, but we still need to upload the metadata file.
             # We also need to upload the spec to the spec cache bucket.
-            if check_connector_image_results.status is StepStatus.SKIPPED and not context.pre_release:
+            if check_connector_image_results.status is StepStatus.SKIPPED:
                 context.logger.info(
                     "The connector version is already published. Let's upload metadata.yaml and spec to GCS even if no version bump happened."
                 )
```

Comment:
Out of scope of this PR - i'm not sure we want to do this for spec? The spec is versioned info, we shouldn't be changing it without versioning

Reply:
> a prerelease can have changes to spec

Are we checking the right "image exists"? the prerelease image shouldn't already exist, which would make us upload the spec

Reply:
Forgot to respond here. 

Given that these run off of `context.docker_image` and docker_image is now properly receiving the [docker_image_tag override for prerelease](https://github.com/airbytehq/airbyte/blob/bnchrch/27854/metadata-prerelease/tools/ci_connector_ops/ci_connector_ops/pipelines/contexts.py#L537) this will check if the prerelease tag is published.

Generally its redundant as it should be a new one everytime as the commit is updated.

But this whole function is a bit confusing, updating that now

Reply:
Im going to do this refactor in a different PR.